tests/test_openjdk.py

```diff
@@ -28,3 +30,39 @@ def test_jdk_version(container, java_version):
     assert (
         container.connection.check_output("echo $JAVA_VERSION") == java_version
     )
+
+@pytest.mark.parametrize(
+    "container_per_test",
+    [
+        OPENJDK_11_CONTAINER,
+        OPENJDK_17_CONTAINER,
+    ],
+    indirect=["container_per_test"]
+)
+@pytest.mark.parametrize(
+    "test_to_run, expected_string",
+    [
+        ("threads_concurrency_and_sleep" , ["I am the thread 1", "I am the thread 2"])
+    ],
+)
+def test_jdk_extended(
+    container_per_test, test_to_run, expected_string, host, container_runtime
+):
+    # copy the test to the container
+    host.run_expect(
+        [0],
+        f"{container_runtime.runner_binary} cp {appdir + test_to_run}.java"
+        f" {container_per_test.container_id}:{condir}",
+    )
+
+    # run the test
+    testout = container_per_test.connection.run_expect(
+        [0], "java " + condir + test_to_run + ".java"
+    )
+
+    # check the results
+    if isinstance(expected_string, list):
+        for check in expected_string:
+            assert check in testout.stdout
+    else:
+        assert expected_string in testout.stdout
```

Comment:
This should not be necessary, you do **know** which type `expected_string` is. Also, if it is a list, then the variable name is not the best :wink:

Reply:
Done